src/test/java/com/rocketden/main/service/ProblemServiceTests.java

```diff
@@ -452,6 +449,30 @@ public void editProblemBadTestCaseOutput() {
 
         assertEquals(ProblemError.INVALID_INPUT, exception.getError());
     }
+    @Test
+    public void editProblemBadApproval() {
+        Problem problem = new Problem();
+        problem.setName(NAME);
+        problem.setDescription(DESCRIPTION);
+        problem.setDifficulty(ProblemDifficulty.MEDIUM);
+        problem.setOutputType(ProblemIOType.STRING);
+        ProblemInput problemInput = new ProblemInput(INPUT_NAME, IO_TYPE);
+        problem.addProblemInput(problemInput);
+
+        Mockito.doReturn(problem).when(repository).findProblemByProblemId(problem.getProblemId());
+
+        ProblemDto updatedProblem = ProblemMapper.toDto(problem);
+        updatedProblem.setApproval(true);
+
+        List<ProblemTestCaseDto> emptyCases = new ArrayList<ProblemTestCaseDto>();
+        updatedProblem.setTestCases(emptyCases);
```

Comment:
Just want to note that this shouldn't be necessary, as the test cases are, by default, set to an empty `ArrayList`.